src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -0,0 +1,161 @@
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Stryker.Core.TestRunners.VsTest
+{
+    public partial class VsTestRunner : ITestRunner
+    {
+        private readonly StrykerOptions _options;
+        private readonly ProjectInfo _projectInfo;
+        private string _defaultRunSettings;
+        private int _timeoutMS = 0;
+        private readonly VsTestHelper _vsTestHelper;
+        private List<string> _messages = new List<string>();
+
+        public VsTestRunner(StrykerOptions options, ProjectInfo projectInfo)
+        {
+            _options = options;
+            _projectInfo = projectInfo;
+            _vsTestHelper = new VsTestHelper(options);
+
+            GenerateRunSettings();
+        }
+
+        public TestRunResult RunAll(int? timeoutMS, int? activeMutationId)
+        {
+            _timeoutMS = timeoutMS ?? 0;
+            GenerateRunSettings();
+
+            var testBinariesPath = FilePathUtils.ConvertPathSeparators(Path.Combine(_options.BasePath, "bin", "Debug", _projectInfo.TargetFramework));
+            var testAssemblyPath = FilePathUtils.ConvertPathSeparators(Path.Combine(testBinariesPath, _projectInfo.TestProjectFileName.Replace("csproj", "dll")));
+
+            var vsTestToolPath = _vsTestHelper.GetCurrentPlatformVsTestToolPath();
+            var vsTestExtensionsPath = _vsTestHelper.GetDefaultVsTestExtensionsPath(vsTestToolPath);
+
+            IVsTestConsoleWrapper consoleWrapper = null;
+
+            consoleWrapper = new VsTestConsoleWrapper(vsTestToolPath, new ConsoleParameters { TraceLevel = DetermineTraceLevel(), LogFilePath = Path.Combine(_options.BasePath, "StrykerOutput", "logs", "vstest-log.txt") });
+
+            consoleWrapper.StartSession();
+            consoleWrapper.InitializeExtensions(new List<string> { testBinariesPath, vsTestExtensionsPath });
+
+            var testCases = DiscoverTests(new List<string>() { testAssemblyPath }, consoleWrapper);
+
+            var testResults = RunAllTests(consoleWrapper, new List<string> { testAssemblyPath }, activeMutationId);
+
+            // For now we need to throw an OperationCanceledException when a testrun has timed out. 
+            // We know the testrun has timed out because we received less test results from the test run than there are test cases in the unit test project.
+            if (testResults.Count() < testCases.Count())
+            {
+                throw new OperationCanceledException();
+            }
+
+            var testResult = new TestRunResult
+            {
+                Success = testResults.All(tr => tr.Outcome == TestOutcome.Passed),
+                ResultMessage = string.Join(
+                    Environment.NewLine,
+                    testResults.Where(tr => !string.IsNullOrWhiteSpace(tr.ErrorMessage)).Select(tr => tr.ErrorMessage)),
+                TotalNumberOfTests = testCases.Count()
+            };
+
+            consoleWrapper.EndSession();
```

Comment:
We should somehow call this when all mutants have been tested. That way we keep the session open for the whole stryker run.